lightningd/options.c

```diff
@@ -697,10 +700,13 @@ static void opt_parse_from_config(struct lightningd *ld)
 	char *argv[3];
 	int i, argc;
 
-	contents = grab_file(ld, "config");
+	if (ld->config_filename!= NULL) {
+		contents = grab_file(ld, ld->config_filename);
+	} else
+		contents = grab_file(ld, "config");
 	/* Doesn't have to exist. */
 	if (!contents) {
-		if (errno != ENOENT)
+		if ((errno != ENOENT) || ld->config_filename != NULL)
```

Comment:
A comment here would be much appreciated: in the default config case we allow it not to exist, but if the config was specified on the command line it has to exist.

Reply:
Also inconsistent parentheses.